main.cc

```diff
@@ -836,6 +836,7 @@ int main(int ac, char** av) {
 
             gms::gossip_config gcfg;
             gcfg.gossip_scheduling_group = dbcfg.gossip_scheduling_group;
+            gcfg.snitch_name = locator::i_endpoint_snitch::get_local_snitch_ptr()->get_name();
```

Comment:
This is not invariant and can be changed in a runtime via REST API: https://github.com/scylladb/scylla/blob/master/api/storage_service.cc#L686

Reply:
I don't see how this is API is any useful. However we may want to either take care of that or remove this API in the context of this change.

Reply:
The only place where gossiper calls snitch for its name is the `check_snitch_matches()`, which in turn is only called on start. So in order for the API to wreak havoc is to get triggerred until the node finishes its shadow round. Is that really _that_ awful?

Reply:
For the sake of correctness AFAICT check_snitch_name_matches() is called from two places: storage_service::check_for_endpoint_collision(...) and storage_service::prepare_to_join(...).

But what if this changes in the future? My point was about a "spaghetti" code we are creating here. If we want to make a Snitch name to be an invariant - let's make it such and I see no issue whatsoever to drop that API which nobody uses as far as I'm aware of.

Reply:
If not dropping the API, let's then add the `snitch_name` argument, all the more so its peer `check_knows_remote_features` accepts checked features as arguments, instead of getting them itself.

Reply:
@vladzcloudius I'm all for removing the API, no code is always better :) But if not, then -- the `check_for_endpoint_collision` is called from storage service, my proposal is to get snitch name in those places, (storage service already gets it [nearby](https://github.com/scylladb/scylla/blob/c38ceafdcfca87b3706b55fe9c73a7f015a4b560/service/storage_service.cc#L349)) and pass it as argument into `check_for_endpoint_collision`.

Reply:
Yup. As long as we don't cache it - we should be safe enough.